test/externalTests/ens.sh

```diff
@@ -65,6 +65,10 @@ function ens_test
     force_hardhat_compiler_settings "$config_file" "$(first_word "$SELECTED_PRESETS")"
     yarn install
 
+    # With ethers.js 5.6.2 many tests for revert messages fail.
+    # TODO: Remove when https://github.com/ethers-io/ethers.js/discussions/2849 is resolved.
+    yarn add ethers@5.6.1
```

Comment:
I don't know `yarn` or `npm` very well, but what is the reason for doing `yarn add ethers@5.6.1` after `yarn install`. The same pattern I can also see later with `npm`. What is the semantic here? That `yarn install`/`install install`  is installing a newer `ether.js`, that's why we overwrite that with an older version?

Reply:
Yes. It's to get specifically version 5.6.1. The `install` command installs whathever `package.json` and `package-lock.json`/`yarn.lock` tell it to. For example for ENS:
```json
"ethers": "^5.1.4",
```
so it installs the latest ethers.js in the 5.x.y series (currently 5.6.2).

One way to override it would be to patch `package.json`. Another is just to install the version we want afterwards. I used to do the former but installing it afterwards is cleaner and less likely to break.